### PR TITLE
Add test vectors for larger payload sizes

### PR DESCRIPTION
**Pull request by Steve132**

### CODE REVIEW

File: spec/cashaddr.md [hunk 1]
```diff
@@ -147,6 +147,37 @@ The following addresses are given in the legacy and new format.
...
+|56|pref:pmvl5lzvdm6km38lgga64ek5jhdl7e3aqd9895wu04fvhlnare5937w4ywkq57juxsrhvw8ym5d8qx7sz7zz0zvcypqsammyqffl|D9FA7C4C6EF56DC4FF423BAAE6D495DBFF663D034A72D1DC7D52CBFE7D1E6858F9D523AC0A7A5C34077638E4DD1A701BD017842789982041|
+|64|bitcoincash:qlg0x333p4238k0qrc5ej7rzfw5g8e4a4r6vvzyrcy8j3s5k0en7calvclhw46hudk5flttj6ydvjc0pv3nchp52amk97tqa5zygg96mtky5sv5w|D0F346310D5513D9E01E299978624BA883E6BDA8F4C60883C10F28C2967E67EC77ECC7EEEAEAFC6DA89FAD72D11AC961E164678B868AEEEC5F2C1DA08884175B|
+|64|bchtest:plg0x333p4238k0qrc5ej7rzfw5g8e4a4r6vvzyrcy8j3s5k0en7calvclhw46hudk5flttj6ydvjc0pv3nchp52amk97tqa5zygg96mc773cwez|D0F346310D5513D9E01E299978624BA883E6BDA8F4C60883C10F28C2967E67EC77ECC7EEEAEAFC6DA89FAD72D11AC961E164678B868AEEEC5F2C1DA08884175B|
+|64|pref:plg0x333p4238k0qrc5ej7rzfw5g8e4a4r6vvzyrcy8j3s5k0en7calvclhw46hudk5flttj6ydvjc0pv3nchp52amk97tqa5zygg96mg7pj3lh8|D0F346310D5513D9E01E299978624BA883E6BDA8F4C60883C10F28C2967E67EC77ECC7EEEAEAFC6DA89FAD72D11AC961E164678B868AEEEC5F2C1DA08884175B|
```
Comment (deadalnix) [inline spec/cashaddr.md]: I was able to check the bitcoincash and bchtest addresses here: https://reviews.bitcoinabc.org/D1456

I changed the pref ones to also test larger version number. I sthat possible to update the pref test vector to reflect the one from the diff ?

**Comment by Steve132** [comment]
It should be pretty easy to regenerate so it should be possible, but I'm
not sure I follow what you mean, exactly by "larger version number" or the
diff.  Can you clarify what you want and I can make those changes?

By "larger version number" do you mean like, 0x5 instead of 0x0? or do you
mean my extension proposal here?

**Comment by deadalnix** [comment]
Sample case in the mentioned diff are using 0x0f as a version.

**Comment by Steve132** [comment]
Do you want me to add the vectors you've added in the diff to the spec?    I see what you mean now, but what are you asking me to do?

**Comment by Steve132** [comment]
Okay, I'll work on that.

**Comment by Steve132** [comment]
I added the changes you requested from your test vectors in your diff

**Comment by schancel** [comment]
Thank you @Steve132 .  Will let @deadalnix sign off.

**Comment by deadalnix** [comment]
The type of all the bchtest addresses is invalid.

**Comment by Steve132** [comment]
I'm not sure I understand what you mean by that.

Okay, you're right.  I forgot that my test generated the bchtest addresses
with a p2sh type bits.  I'm updating the table now to indicate that, but I
could also regenerate the addresses if you prefer.

**Comment by deadalnix** [comment]
Merged, thanks.

File: spec/cashaddr.md [hunk 2]
```diff
@@ -147,6 +147,48 @@ The following addresses are given in the legacy and new format.
...
+## Larger Test Vectors
+
+This table defines test vectors for various payloads of sizes 160-512 bits with various prefixes.   These test vectors aren't given in legacy address format because the legacy format is limited to payloads of 160 bits.
+
+
+| Payload Size (bytes) | Type | CashAddr | Payload (hex) |
+|:----------------------|:--|:----------|:---------------|
+|20|0|bitcoincash:qr6m7j9njldwwzlg9v7v53unlr4jkmx6eylep8ekg2|F5BF48B397DAE70BE82B3CCA4793F8EB2B6CDAC9|
+|20|1|bchtest:pr6m7j9njldwwzlg9v7v53unlr4jkmx6eyvwc0uz5t|F5BF48B397DAE70BE82B3CCA4793F8EB2B6CDAC9|
+|20|0|pref:pr6m7j9njldwwzlg9v7v53unlr4jkmx6ey65nvtks5|F5BF48B397DAE70BE82B3CCA4793F8EB2B6CDAC9|
```
Comment (braydonf) [inline spec/cashaddr.md]: The version byte should be the same here as line 157, no? As the 'Payload Size' and the 'Type' are the same. The closest I've found to these vectors in tests is at https://github.com/Bitcoin-ABC/bitcoin-abc/blob/master/src/test/cashaddrenc_tests.cpp#L310 and this isn't included. Is this meant to be invalid?

**Pull request merged**